ProcessLib/BoundaryCondition/NonuniformVariableDependentNeumannBoundaryConditionLocalAssembler.h

```diff
@@ -0,0 +1,131 @@
+          _local_matrix_size(local_matrix_size)
+    {
+    }
+
+    void assemble(std::size_t const mesh_item_id,
+                  NumLib::LocalToGlobalIndexMap const& dof_table_boundary,
+                  double const t, const GlobalVector& x, GlobalMatrix& /*K*/,
+                  GlobalVector& b, GlobalMatrix* /*Jac*/) override
+    {
+        NodalVectorType _local_rhs(_local_matrix_size);
+        _local_rhs.setZero();
+        MeshNodeParameter<double> constant_values{"ConstantValues",
+                                                  _data.constant};
+        MeshNodeParameter<double> coefficient_current_variable_values{
+            "CurrentVariableValues", _data.coefficient_current_variable};
+        MeshNodeParameter<double> coefficient_other_variable_values{
+            "OtherVariableValues", _data.coefficient_other_variable};
+        MeshNodeParameter<double> coefficient_mixed_variables_values{
+            "MixedVariablesValues", _data.coefficient_mixed_variables};
+        // Get element nodes for the interpolation from nodes to
+        // integration point.
+        NodalVectorType const constant_node_values =
+            constant_values.getNodalValuesOnElement(Base::_element, t);
+        NodalVectorType const coefficient_current_variable_node_values =
+            coefficient_current_variable_values.getNodalValuesOnElement(
+                Base::_element, t);
+        NodalVectorType const coefficient_other_variable_node_values =
+            coefficient_other_variable_values.getNodalValuesOnElement(
+                Base::_element, t);
+        NodalVectorType const coefficient_mixed_variables_node_values =
+            coefficient_mixed_variables_values.getNodalValuesOnElement(
+                Base::_element, t);
+        unsigned const n_integration_points = _local_matrix_size;
+
+        auto const indices_current_variable =
+            NumLib::getIndices(mesh_item_id, dof_table_boundary);
+        auto const indices_other_variable = NumLib::getIndices(
+            mesh_item_id, _data.dof_table_boundary_other_variable);
+        std::vector<double> const local_current_variable =
+            x.get(indices_current_variable);
+        std::vector<double> const local_other_variable =
+            x.get(indices_other_variable);
+
+        for (unsigned ip = 0; ip < n_integration_points; ip++)
+        {
+            auto const& n_and_weight = Base::_ns_and_weights[ip];
+            auto const& N = n_and_weight.N;
+            auto const& w = n_and_weight.weight;
+
+            double current_variable_int_pt = 0.0;
+            double other_variable_int_pt = 0.0;
+
+            NumLib::shapeFunctionInterpolate(local_current_variable, N,
+                                             current_variable_int_pt);
+            NumLib::shapeFunctionInterpolate(local_other_variable, N,
+                                             other_variable_int_pt);
+            NodalVectorType const neumann_node_values =
```

Comment:
Maybe you can put some words in the parameter documentation of NonuniformVariableDependentNeumann like
"If the Newton-Raphson method is applied for the global system, please select the numerical Jacobian calculation.".